analysers/analyser_osmosis_building_in_polygon.py

```diff
@@ -24,23 +24,48 @@
 from .Analyser_Osmosis import Analyser_Osmosis
 
 sql10 = """
+CREATE TEMP TABLE poly_landuse AS
 SELECT
-  DISTINCT ON (ways.id)
-  ways.id,
+  type_id,
+  poly_proj,
+  tags->'landuse' AS landuse
+FROM (
+  SELECT
+    'W' || id AS type_id,
+    tags,
+    ST_Transform(ST_MakePolygon(ways.linestring), {proj}) AS poly_proj
+  FROM
+    ways
+  WHERE
+    is_polygon AND
+    tags != ''::hstore
+  UNION ALL
+  SELECT
+    'R' || id AS type_id,
+    tags,
+    poly_proj
+  FROM
+    multipolygons
+  WHERE
+    is_valid
+) AS t
+WHERE
+  tags?'landuse' AND
+  tags->'landuse' IN ('farmland', 'vineyard', 'orchard', 'plant_nursery')
+"""
+
+sql11 = """
+SELECT DISTINCT ON (poly_landuse.type_id)
```

Comment:
`DISTINCT ON` should come with an `ORDER BY`.

Reply:
Right, otherwise the results would be random.
It was missing in the initial implementation too